builtin/providers/aws/resource_aws_api_gateway_base_path_mapping.go

```diff
@@ -0,0 +1,130 @@
+package aws
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsApiGatewayBasePathMapping() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsApiGatewayBasePathMappingCreate,
+		Read:   resourceAwsApiGatewayBasePathMappingRead,
+		Delete: resourceAwsApiGatewayBasePathMappingDelete,
+
+		Schema: map[string]*schema.Schema{
+			"api_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"base_path": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"stage_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"domain_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsApiGatewayBasePathMappingCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	err := resource.Retry(30*time.Second, func() *resource.RetryError {
+		_, err := conn.CreateBasePathMapping(&apigateway.CreateBasePathMappingInput{
+			RestApiId:  aws.String(d.Get("api_id").(string)),
+			DomainName: aws.String(d.Get("domain_name").(string)),
+			BasePath:   aws.String(d.Get("base_path").(string)),
+			Stage:      aws.String(d.Get("stage_name").(string)),
+		})
+
+		if err != nil {
+			if err, ok := err.(awserr.Error); ok && err.Code() != "BadRequestException" {
+				return &resource.RetryError{
+					Err:       err,
+					Retryable: false,
+				}
+			}
+
+			return &resource.RetryError{
+				Err:       fmt.Errorf("Error creating Gateway base path mapping: %s", err),
+				Retryable: true,
+			}
+		}
+
+		id := fmt.Sprintf("%s/%s", d.Get("domain_name").(string), d.Get("base_path").(string))
+		d.SetId(id)
```

Comment:
I think it would make more sense to keep these 2 lines outside of the retry block, but it's really a nitpick! 😃